### PR TITLE
S3 browser more general code cleanup

### PR DESCRIPTION
**Pull request by hunterwerlla**
- Merge the two copies of `S3Resources.kt`
- Fix reading file into memory in upload
- fix several unnecessary `var` to `val`
- Move `CreateBucketAction` into `bucketActions`
- Move `S3RowSorter` and `S3VirtualBucket` into the `bucketEditor` directory

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/S3ExplorerNode.kt [hunk 1]
```diff
@@ -43,7 +44,9 @@ class S3BucketNode(project: Project, val bucket: Bucket) :
             val editorManager = FileEditorManager.getInstance(nodeProject)
             // See if there is already an open editor, otherwise make a new one
             val virtualFile =
-                editorManager.openFiles.firstOrNull { (it as? S3VirtualBucket)?.s3Bucket?.equals(bucket) == true } ?: S3VirtualBucket(bucket)
+                editorManager.openFiles.firstOrNull { (it as? S3VirtualBucket)?.s3Bucket?.equals(bucket) == true } ?: S3VirtualBucket(
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/S3ExplorerNode.kt]: UNdo this

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/bucketActions/OpenBucketViewerAction.kt [hunk 2]
```diff
@@ -37,6 +37,8 @@ class OpenBucketViewerAction : SingleResourceNodeAction<S3BucketNode>(message("s
 
     private fun openEditor(selected: S3BucketNode, project: Project) {
         val editorManager = FileEditorManager.getInstance(project)
-        editorManager.openTextEditor(OpenFileDescriptor(project, S3VirtualBucket(selected.bucket)), true)
+        editorManager.openTextEditor(OpenFileDescriptor(project,
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/bucketActions/OpenBucketViewerAction.kt]: undo this

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt [hunk 3]
```diff
@@ -14,7 +14,7 @@ import software.amazon.awssdk.services.s3.model.DeleteObjectsRequest
 import software.amazon.awssdk.services.s3.model.ObjectIdentifier
 import software.aws.toolkits.jetbrains.components.telemetry.ActionButtonWrapper
 import software.aws.toolkits.jetbrains.core.AwsClientManager
-import software.aws.toolkits.jetbrains.services.s3.S3VirtualBucket
+import software.aws.toolkits.jetbrains.services.s3.bucketEditor.S3VirtualBucket
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt]: we could probably have called `bucketEditor` just `editor` inside the context of an `s3` package it's pretty obviously going to be bucket-bound.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/UploadObjectAction.kt [hunk 4]
```diff
@@ -73,12 +73,10 @@ class UploadObjectAction(
             .key(key)
             .build()
 
-        val fileChosenSize = fileChosen.inputStream.readBytes().size
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/UploadObjectAction.kt]: I saw this before too and deleted it when I was poking about -- oops. :P

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt (hunk 3 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt]: good idea, renamed

**Pull request merged**